examples/websocket-service-error-handling/websocket_service_error_handling.md

```diff
@@ -0,0 +1,16 @@
+# WebSocket service - Error handling
+
+The Ballerina `websocket` module allows returning errors from the `get` resource and `remote` methods. In the initial WebSocket upgrade service, you can do validations like origin checking and cancel the WebSocket handshake by returning an error from the `get` resource if the validation fails. Once the connection is upgraded to a WebSocket connection, if an error is returned from a remote method in the `websocket:Service`, the error is printed on the server terminal along with the stack trace. Use `do on fail` to catch and handle the error.
```

Comment:
The Ballerina `websocket` module allows returning errors from the `get` resource and `remote` methods. Initial WebSocket upgrade service can do various validations. For instance, origin checking and canceling the WebSocket handshake by returning an error from the `get` resource if the validation fails. In this case it results in `xxx xxxxxxxxxxxxx` HTTP response. Once the connection is upgraded to a WebSocket connection, if an error is returned from a remote method in the `websocket:Service`, the error is printed on the server terminal along with the stack trace and it continues to work. Use `do/fail` to catch and handle the error when needed.